pkg/p2p/libp2p/libp2p.go

```diff
@@ -317,7 +326,11 @@ func (s *Service) Connect(ctx context.Context, addr ma.Multiaddr) (overlay swarm
 		return swarm.Address{}, err
 	}
 
-	s.peers.add(stream.Conn(), i.Address)
+	if s.peers.addIfNotExists(stream.Conn(), i.Address) {
```

Comment:
see the other comment about a similar line. also, i'm not sure that the `trace` line is needed at all